src/registries/npm-registry.js

```diff
@@ -72,10 +73,22 @@ export default class NpmRegistry extends Registry {
     return name.replace('/', '%2f');
   }
 
+  getRequestUrl(registry: string, pathname: string): string {
+    const isPathnameUrl = pathname.match(/^https?:/);
```

Comment:
The `.match` string method is used throughtout this class and I'm guessing those should be repalced as well. I can fix this in another PR.

I will rename the variable to `isUrl`. I think `pathname` can also be a path, `/foo/bar/`, as well as a URL.